fennel/datasets/datasets.py

```diff
@@ -2144,3 +2216,18 @@ def visitFirst(self, obj) -> DSSchema:
             )
         output_schema.name = f"'[Pipeline:{self.pipeline_name}]->first node'"
         return output_schema
+
+    def visitWindow(self, obj) -> DSSchema:
+        output_schema = copy.deepcopy(obj.dsschema())
+        output_schema_name = f"'[Pipeline:{self.pipeline_name}]->window node'"
+
```

Comment:
Lets add a schema check for gap? I think we rely on the to proto to currently ?

Reply:
That is the behavior in join operator but for Window I have called the function which converts string to Timedelta in initiation itself, it will throw the at that time only. Or I can call the function in Schema Validator but we shouldn't assign obj params in Schema Validator, so didn't add there. Anyhow I have written the test for this exactly.